### DISCUSSION TITLE
PMbuild can't find Fortran

### DISCUSSION THREAD

**Discussion by Pagdeklaver** [LAPKB/Pmetrics#202] [Q&A] [answered]
Hi, 
Im trying to install Pmetrics on Mac OS M2 with R studio. The installation works fine, but I have trouble with PMbuild(). It says it can't detect gFortran or homebrew. When I check both in terminal, gfortran and homebred should be installed:
**% gfortran --version
GNU Fortran (Homebrew GCC 13.2.0) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.**

> PMbuild()
Pmetrics cannot find required compiled binary files.
Pmetrics cannot detect gfortran and will attempt to download and install all components.
Do you agree? (Y/N)y
Checking for Brew 
Pmetrics did not find the Homebrew Package Manager and will attempt to download and install it.
Do you wish to proceed? (Y/N)y
A new terminal will open and ask for permission to install.  Enter your computer password.
Press any key when Homebrew installation is complete.

sh: /opt/homebrew/bin/brew: No such file or directory
Pmetrics did not find GCC and will install it.
sh: /opt/homebrew/bin/brew: No such file or directory
ERROR: Pmetrics did not install gfortan automatically.
Please install gfortran manually and then run PMbuild().
Go to http://www.lapk.org/Pmetrics_install.php for help.
ERROR: Pmetrics did not install gfortran automatically.
Please install gfortran manually and then run PMbuild().
Go to http://www.lapk.org/Pmetrics_install.php for help.
Warning messages:
1: In system(paste0(brew_path, " ls --versions gcc")) :
  error in running command
2: In system(paste0(brew_path, " install gcc")) : error in running command

**Comment by mhovd** [comment]
Have you tried to restart RStudio? Especially on Windows this is sometimes necessary as the `%PATH%` does not update for RStudio. While it shouldn't be necessary on Mac, I've found that it sometimes helps just the same.  To identify the cause, would you be so kind as to provide the output of the following commands?

- `which brew`
- `which gfortran`

Thanks!

**Reply by Pagdeklaver** [reply]
Thank you for your reply!

which brew: 
/usr/local/bin/brew
which Fortran:
/usr/local/bin/gfortran

**Reply by mhovd** [reply, marked as answer]
That is curious, usually `/usr/local` is for Intel binaries, while `/opt/homebrew/ ` is for ARM, i.e. Apple chips M1 and up. We will review the install script and try to find the issue. Still, let's try to get you running Pmetrics before we can publish the fix.

Try installing brew again, with `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`
NOTE: After the install, brew may ask you to copy and paste two commands into your terminal.

Then, install `gcc` with `brew install gcc`.

Then close RStudio and Terminal, and re-open terminal, running the same commands as before, i.e.

- `which brew`
- `which gfortran`

Pmetrics expects these to live in `/opt/homebrew/bin/`, which is the root of the issue. Let me know if this doesn't work.

**Reply by Pagdeklaver** [reply]
thank you! I don't have an error on installation anymore. I think this problem originated from the installation of my macbook: I migrated from Intel macbook to M2 macbook and assume the wrong directory was used.

**Reply by mhovd** [reply]
I'm happy to hear! If you migrated from an Intel MacBook that makes perfect sense.

**Comment by Siel** [comment]
Hi @Pagdeklaver, thanks for the report.

As @mhovd was telling you, when Pmetrics detects an ARM based MacOS system, it expects gfotran to be installed under `/opt/homebrew/bin`, given that because some external circumstances this might not be the case in your system, you can override this default by setting the right Pmetrics option.

The way of doing this is using `setPMoptions()`, there is an option there named `gfortran_path` (setting that might also change the `compilation_statements` that is the command used to compile the Fortran binaries). Using `setPMoptions(gfortran_path=<YOUR_GFOTRAN_PATH>)` might solve the issue.

Since your Rstudio might be running under Rosetta, a handy way to detect where located the gfortran your R installation might have access to is by executing `system("which gfortran")` in your R terminal.

Hope this works!
